presto-main/src/main/java/com/facebook/presto/operator/scalar/MapFilterFunction.java

```diff
@@ -195,9 +209,19 @@ private static MethodHandle generateFilter(Type keyType, Type valueType, Type ma
                                         .append(keySqlType.invoke("appendTo", void.class, block, position, blockBuilder))
                                         .append(valueSqlType.invoke("appendTo", void.class, block, add(position, constantInt(1)), blockBuilder))))));
 
-        body.append(blockBuilder.invoke("build", Block.class).ret());
+        body.append(mapBlockBuilder
+                .invoke("closeEntry", BlockBuilder.class)
```

Comment:
We do. `map_filter` doesn't throw exceptions. However, you're right in that the evaluation of the lambda expression could throw exceptions. This issue also applies to other map functions. This need to be addressed separately.